src/ephemeris.c

```diff
@@ -156,27 +160,29 @@ double predict_range(double rx_pos[3],
   double temp[3];
   double clock_err, clock_rate_err;
 
-  calc_sat_pos(sat_pos, sat_vel, &clock_err, &clock_rate_err, ephemeris, tot);
+  calc_sat_state(ephemeris, t, sat_pos, sat_vel, &clock_err, &clock_rate_err);
 
-  vector_subtract(3, sat_pos, rx_pos, temp); // temp = sat_pos - rx_pos
+  vector_subtract(3, sat_pos, rx_pos, temp); /* temp = sat_pos - rx_pos */
   return vector_norm(3, temp);
 }
 
-/** Do we have an ephemeris for this PRN?
- * \TODO  This should actually be more than just the "valid" flag.
- *         When we write an is_usable() function, lets use that instead
- *         of just es[prn].valid.
+/** Is this ephemeris usable?
  *
- * \param prn     The prn we want to check.
+ * \todo This should actually be more than just the "valid" flag.
+ *       When we write an is_usable() function, lets use that instead
+ *       of just es[prn].valid.
```

Comment:
In a PR of mine somewhere, there's an added "healthy" flag. Might be bundled in with the sat set handling changes that fixed the LD issue..